pkg/river/vm/vm.go

```diff
@@ -506,6 +506,9 @@ func (vm *Evaluator) evaluateExpr(scope *Scope, assoc map[value.Value]ast.Node,
 
 // A Scope exposes a set of variables available to use during evaluation.
 type Scope struct {
+	// Name is the name of this particular scope/block
+	Name string
```

Comment:
Sorry, why do we want to name the scopes?

Reply:
It's mostly debugging, to figure out which one we are on.

Reply:
I'm not sure about this IMO, every component effectively has the same scope 🤔

Reply:
Isn't each component a separate scope?

Reply:
No, the scope is the object-ized form of the entire graph, each component gets a copy of it at evaluation time.

Reply:
That's true, it was helpful when I was trying to dig through the code to see what part of the evaluation I was at but I can walk the stack to see that.